pandas/io/formats/style.py

```diff
@@ -464,13 +464,26 @@ def _update_ctx(self, attrs: DataFrame) -> None:
             Whitespace shouldn't matter and the final trailing ';' shouldn't
             matter.
         """
+        err = KeyError(
+            "`Styler.apply` and `.applymap` are not compatible with subset slices "
+            "containing non-unique index or column keys."
+        )
+
         for cn in attrs.columns:
-            for rn, c in attrs[[cn]].itertuples():
-                if not c:
-                    continue
-                css_list = maybe_convert_css_to_tuples(c)
-                i, j = self.index.get_loc(rn), self.columns.get_loc(cn)
-                self.ctx[(i, j)].extend(css_list)
+            try:
+                for rn, c in attrs[[cn]].itertuples():
+                    if not c:
+                        continue
+                    css_list = maybe_convert_css_to_tuples(c)
+                    i, j = self.index.get_loc(rn), self.columns.get_loc(cn)
```

Comment:
can instead we just not use the indices to look up locations? and instead just use indexers (e.g. iterate over the number of columns and use iloc)

Reply:
no, because `attrs` may be a `subset` of the main `self.data`, so to maintain performance and ensure the mapped css is placed in the right integer locations it needs to lookup - and lookup doesn't work (without ambiguity) for non-unique.